project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/UserDataFixture.php

```diff
@@ -31,48 +36,102 @@ class UserDataFixture extends AbstractReferenceFixture implements DependentFixtu
+     * @param \Shopsys\FrameworkBundle\Model\Customer\UserDataFactoryInterface $userDataFactory
      */
     public function __construct(
         CustomerFacade $customerFacade,
-        UserDataFixtureLoader $loaderService,
         Generator $faker,
         EntityManagerInterface $em,
-        HashGenerator $hashGenerator
+        HashGenerator $hashGenerator,
+        Domain $domain,
+        CustomerDataFactoryInterface $customerDataFactory,
+        UserDataFactoryInterface $userDataFactory
     ) {
         $this->customerFacade = $customerFacade;
-        $this->loaderService = $loaderService;
         $this->faker = $faker;
         $this->em = $em;
         $this->hashGenerator = $hashGenerator;
+        $this->domain = $domain;
+        $this->customerDataFactory = $customerDataFactory;
+        $this->userDataFactory = $userDataFactory;
     }
 
     /**
      * @param \Doctrine\Common\Persistence\ObjectManager $manager
      */
     public function load(ObjectManager $manager)
     {
+        /** @var \Shopsys\FrameworkBundle\Model\Country\Country[] $countries */
         $countries = [
-            $this->getReference(CountryDataFixture::COUNTRY_CZECH_REPUBLIC),
-            $this->getReference(CountryDataFixture::COUNTRY_SLOVAKIA),
+            self::FIRST_DOMAIN_ID => $this->getReference(CountryDataFixture::COUNTRY_CZECH_REPUBLIC),
+            self::SECOND_DOMAIN_ID => $this->getReference(CountryDataFixture::COUNTRY_SLOVAKIA),
         ];
-        $this->loaderService->injectReferences($countries);
-
-        $customersData = $this->loaderService->getCustomersDataByDomainId(Domain::FIRST_DOMAIN_ID);
-
-        foreach ($customersData as $customerData) {
-            $customerData->userData->createdAt = $this->faker->dateTimeBetween('-1 week', 'now');
-
-            $customer = $this->customerFacade->create($customerData);
 
-            if ($customer->getId() === 1) {
-                $this->resetPassword($customer);
-                $this->addReference(self::USER_WITH_RESET_PASSWORD_HASH, $customer);
+        $fakerSeed = self::FAKER_SEED;
+
+        $domainConfigs = $this->domain->getAll();
+        foreach ($domainConfigs as $domainConfig) {
+            $this->faker->seed($fakerSeed++);
+            $domainId = $domainConfig->getId();
+            for ($customerCount = 0; $customerCount < self::USERS_PER_DOMAIN; $customerCount++) {
+                $userData = $this->userDataFactory->createForDomainId($domainId);
+
+                $userData->domainId = $domainId;
+                $userData->firstName = $this->faker->firstName;
+                $userData->lastName = $this->faker->lastName;
+                $userData->createdAt = $this->faker->dateTimeBetween('-1 week', 'now');
+                $userData->email = $userData->firstName . '.' . $userData->lastName . '@no-reply.com';
+                $userData->password = $this->faker->password;
```

Comment:
is faked password needed? doesn't make more sense to put there the same password for anyone?